lib/db/instance.go

```diff
@@ -177,25 +263,37 @@ func (db *instance) withHaveSequence(folder []byte, startSeq int64, fn Iterator)
 			}
 		}
 		if !fn(f) {
-			return
+			return nil
 		}
 	}
+	return dbi.Error()
 }
 
-func (db *instance) withAllFolderTruncated(folder []byte, fn func(device []byte, f FileInfoTruncated) bool) {
-	t := db.newReadWriteTransaction()
+func (db *instance) withAllFolderTruncated(folder []byte, fn func(device []byte, f FileInfoTruncated) bool) error {
+	t, err := db.newReadWriteTransaction()
+	if err != nil {
+		return err
+	}
 	defer t.close()
 
-	dbi := t.NewIterator(util.BytesPrefix(db.keyer.GenerateDeviceFileKey(nil, folder, nil, nil).WithoutNameAndDevice()), nil)
+	key, err := db.keyer.GenerateDeviceFileKey(nil, folder, nil, nil)
+	if err != nil {
+		return err
+	}
+	dbi, err := t.NewPrefixIterator(key.WithoutNameAndDevice())
+	if err != nil {
+		return err
+	}
 	defer dbi.Release()
 
 	var gk, keyBuf []byte
 	for dbi.Next() {
 		device, ok := db.keyer.DeviceFromDeviceFileKey(dbi.Key())
 		if !ok {
 			// Not having the device in the index is bad. Clear it.
-			t.Delete(dbi.Key())
-			t.checkFlush()
+			if err := t.Delete(dbi.Key()); err != nil {
```

Comment:
release/close don't return errors. They are deferred to ensure cleanup in case we return early due to some other error. The method then ends with `return commit()` which does pass the final error.